instrumentation/pg/test/opentelemetry/instrumentation/pg/instrumentation_test.rb

```diff
@@ -361,5 +373,21 @@
         _(span.attributes['net.peer.port']).must_equal port.to_i if PG.const_defined?(:DEF_PORT)
       end
     end
+
+    def self.load_fixture
+      data = File.read("#{Dir.pwd}/test/fixtures/sql_table_name.json")
+      JSON.parse(data)
+    end
+
+    load_fixture.each do |test_case|
+      name = test_case['name']
+      query = test_case['sql']
+
+      define_method(:"test_sql_table_name_#{name}") do
+        table_name = client.send(:collection_name, query)
+
+        assert('test_table', table_name)
+      end
+    end
```

Comment:
Did you mean `assert_equal(expected, actual)` here? `assert()` tests the truthiness of the first argument which is currently always the string literal `"test_table"` so this assertion as written always passes.

I've made 3 changes in the unfortunately-noisy suggestion below:

1. Wrapped these fixture-driven tests in a `describe` block. 

The dynamically-defined tests were getting added to multiple other minispec contexts. This would generate multiple failures from a single test failure and the output would be confusingly named. For example:

```
1) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation
  ::tracing
  ::when connection has multiple hosts#test_sql_table_name_drop_table
  [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
```

2. Changed the assertion from `assert()` to `assert_equal()`.

```diff
- assert('test_table', table_name)
+ assert_equal('test_table', table_name)
```

<details><summary>11 errors:</summary>

```
  1) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_create_table [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

  2) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_from_with_join [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table1"

  3) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_drop_table [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: nil

  4) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_delete_from [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

  5) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_insert_into [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "X"

  6) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_alter_table [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: nil

  7) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_from_with_subquery [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

  8) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_create_table_if_not_exists [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

  9) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_drop_table_if_exists [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

 10) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_update [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"

 11) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_select_count_from [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "test_table"
  Actual: "table_name"
```

</details>

3. Changed the expected table name to "table_name":

"table_name" was the name used for most of the queries, so the error count went from 11 to 5.

```diff
- assert_equal('test_table', table_name)
+ assert_equal('table_name', table_name)
```

<details><summary>Errors went from 11 to 5</summary>

```
  1) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_insert_into [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "table_name"
  Actual: "X"

  2) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_from [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "table_name"
  Actual: "test_table"

  3) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_drop_table [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "table_name"
  Actual: nil

  4) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_alter_table [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "table_name"
  Actual: nil

  5) Failure:
OpenTelemetry::Instrumentation::PG::Instrumentation::tracing::#connection_name#test_sql_table_name_from_with_join [test/opentelemetry/instrumentation/pg/instrumentation_test.rb:390]:
Expected: "table_name"
  Actual: "table1"
```

</details>


```suggestion
    describe '#connection_name' do
      def self.load_fixture
        data = File.read("#{Dir.pwd}/test/fixtures/sql_table_name.json")
        JSON.parse(data)
      end

      load_fixture.each do |test_case|
        name = test_case['name']
        query = test_case['sql']

        define_method(:"test_sql_table_name_#{name}") do
          table_name = client.send(:collection_name, query)

          assert_equal('table_name', table_name) # TODO: use an expected name from fixtures or update fixtures to always use "table_name"
        end
      end
    end
```